plugin/ale.vim

```diff
@@ -70,6 +70,20 @@ if g:ale_lint_on_save
     augroup END
 endif
 
+" This flag can be set to 1 to enable linting when leaving insert mode.
+let g:ale_lint_in_normal_mode = get(g:, 'ale_lint_in_normal_mode', 0)
+if g:ale_lint_in_normal_mode
+    augroup ALERunOnInsertLeaveGroup
+        autocmd!
+        autocmd InsertLeave * call ale#Queue(0)
+    augroup END
+
+    augroup ALERunOnTextChangedNormalGroup
+        autocmd!
+        autocmd TextChanged * call ale#Queue(g:ale_lint_delay)
+    augroup END
```

Comment:
I wonder if it will be confusing that you can set the option for linting when text is changed to being off, but then have it so it's still linting when text is changed. I wonder if this `autocmd` should be removed.

Another option would be to split the option into three options, for toggling linting in normal mode, insert mode, and when leaving insert mode, with the old option kept for turning the first two off easily.